gui/packages/ubuntupro/lib/pages/subscribe_now/subscribe_now_model.dart

```diff
@@ -35,8 +52,29 @@ class SubscribeNowModel {
     }
   }
 
-  /// Returns true if the environment variable 'UP4W_ALLOW_STORE_PURCHASE' has been set.
-  /// Since this reading won't change during the app lifetime, even if the user changes
-  /// it's value from outside, the value is cached so we don't check the environment more than once.
-  bool get purchaseAllowed => _isPurchaseAllowed;
+  void tokenUpdate(String token) async {
+    _token.update(token);
+    notifyListeners();
+  }
```

Comment:
If you accept removing the ProTokenValue class but still use eithers you can inline this:

```suggestion
  void tokenUpdate(String raw) async {
    _token = ProToken.create(raw);
    notifyListeners();
  }
```

Reply:
Similarly, the model itself could have a `clear()` method:

```dart
void clear() {
    _token = const Either.right(null);
  }
```